packages/react-badge/src/components/Badge/Badge.types.ts

```diff
@@ -5,39 +5,39 @@ export type BadgeSlots = {
   icon?: Slot<'span'>;
 };
 
-export type BadgeCommons = {
+// react has a non-standard `color` attribute in its types
+// https://github.com/DefinitelyTyped/DefinitelyTyped/blob/a4ab0fa432320e70da9e51c8ae2e47377f65804b/types/react/index.d.ts#L1868
+export type BadgeProps = Omit<ComponentProps<BadgeSlots>, 'color'> & {
   /**
    * A Badge can be filled, outline, ghost, inverted
    * @defaultvalue filled
    */
-  appearance: 'filled' | 'ghost' | 'outline' | 'tint';
+  appearance?: 'filled' | 'ghost' | 'outline' | 'tint';
 
   /**
    * A Badge can be one of preset colors
    * @defaultvalue brand
    */
-  color: 'brand' | 'danger' | 'important' | 'informative' | 'severe' | 'subtle' | 'success' | 'warning';
+  color?: 'brand' | 'danger' | 'important' | 'informative' | 'severe' | 'subtle' | 'success' | 'warning';
 
   /**
    * A Badge can position the icon before or after the content.
    * @defaultvalue before
    */
-  iconPosition: 'before' | 'after';
+  iconPosition?: 'before' | 'after';
 
   /**
    * A Badge can be square, circular or rounded.
    * @defaultvalue circular
    */
-  shape: 'circular' | 'rounded' | 'square';
+  shape?: 'circular' | 'rounded' | 'square';
 
   /**
    * A Badge can be on of several preset sizes.
    * @defaultvalue medium
    */
-  size: 'tiny' | 'extra-small' | 'small' | 'medium' | 'large' | 'extra-large';
+  size?: 'tiny' | 'extra-small' | 'small' | 'medium' | 'large' | 'extra-large';
 };
 
-// react has a non-standard `color` attribute in its types
-// https://github.com/DefinitelyTyped/DefinitelyTyped/blob/a4ab0fa432320e70da9e51c8ae2e47377f65804b/types/react/index.d.ts#L1868
-export type BadgeProps = Omit<ComponentProps<BadgeSlots>, 'color'> & Partial<BadgeCommons>;
-export type BadgeState = ComponentState<BadgeSlots> & BadgeCommons;
+export type BadgeState = ComponentState<BadgeSlots> &
+  Required<Pick<BadgeProps, 'appearance' | 'color' | 'iconPosition' | 'shape' | 'size'>>;
```

Comment:
```suggestion
  Required<BadgeProps>;
```
nit: you shouldn't need to use `Pick`

Reply:
You do not need to use it necessarily but if you don't then you'll get all of the HTML props that come from `ComponentProps<BadgeSlots>`.

Reply:
ah yeah, good point. IMO this was one of the advantages of `Commons` types since it the props would exclude all HTML props by default 🤔🤔🤔

Reply:
Yup. Badge is a little bit of an outlier, since almost all of the props do end up also going on state since they're essentially all for styling. I chose `Pick` instead of `Omit` because I still find it more explicit/easier to read, but for Badge it could go either way.